File: server/internal/models/budget.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Budget struct {
	gorm.Model
	Name        string    `gorm:"not null" json:"name"`
	TotalAmount float64   `gorm:"not null" json:"totalAmount"`
	StartDate   time.Time `json:"startDate"`
	EndDate     time.Time `json:"endDate"`
	UserID      uint      `gorm:"not null" json:"userId"`
	Expenses    []Expense `gorm:"foreignKey:BudgetID" json:"expenses"`
}

File: client/app/interface/budget.ts
import { Expense } from "./expense";

export interface Budget {
  ID: number;
  name: string;
  totalAmount: number;
  startDate: string;
  endDate: string;
  userId: string;
  expenses: Expense[];
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt?: string | null;
}
